



<p align="center">
  <img src="https://user-images.githubusercontent.com/110712935/269152837-ff947371-ff16-4d1d-aa67-5c325f7d3a01.gif" width="800" height="500" >
</p>

**INSPIRATION**
</br>
The coding DNA of a genome describes the proteins of the organism in terms of 64 different codons that map to roughly 20 different amino acids.  We demonstrate that genomic and evolutionary features can be learned using **Machine Learning (ML)** methods and used for identifying phylogeny and DNA-type of genome-wide coding domains. Further analysis allows for the identification of which codons are most influential for phylogenetic prediction.  Sequencing entire genomes is increasingly easy with next-generation sequencing techniques that are gradually becoming inexpensive


![SEO-DNA-Images-Codons-2019-01-09-12-12-20](https://github.com/21cs043kanivarshini/GitHub/assets/110712935/01fc1194-09bc-46c6-8baa-eac854b6caf6)


Support Vector Machine (SVM) achieved an accuracy of 79.54%.

Logistic Regression yielded an accuracy of 82.59%.

Random Forest (RF) performed well with an accuracy of 88.345%.

XGBoost demonstrated its superiority with an accuracy of 75.45%.




<p align="center">
  <img src="https://github.com/kamesh0407/stock_market_prediction/blob/main/InteloneAPI.jpg" width="400" height="240" >
</p>




Implement DNA codon prediction algorithms using Python.
Optimize algorithms for parallel execution using Intel OneAPI tools.
**Intel OneAPI Integration:**
</br>
Utilize Intel's OneAPI toolkit to the oneAPI cross-architecture programming model provides developers with tools that aim to improve the productivity and velocity of code development when building cross-architecture applications.

**Performance Analysis:**
</br>
* Benchmark and profile the application to measure speedup and resource utilization.
* Identify performance bottlenecks and fine-tune code for better efficiency.

**Visualization and Reporting:**
</br?
* Develop visualization tools to present the predicted codons and analysis results.
* Generate reports summarizing the predictions and performance improvements.

**Technologies and Tools:**
</br>
* Intel OneAPI Toolkit:* Utilize Intel's OneAPI toolkit to the oneAPI cross-architecture programming model
* Programming Languages:* Python for algorithm implementation.
* Data Handling:* Libraries like NumPy and pandas for efficient data handling.
* Visualization:* Matplotlib or other visualization libraries.

**Expected Outcomes:**
</br>
An efficient DNA codon prediction tool optimized for Intel architectures. Improved performance compared to non-accelerated implementations. Visualization tools and reports for easy interpretation of predictions and performance results.

**Benefits:**
</br>
Accelerated DNA codon prediction for genomics research and medical applications. Enhanced understanding of parallel programming and optimization with Intel OneAPI. Contribution to bioinformatics and genomics research.

**Challenges:**
</br>
* Learning curve for utilizing Intel OneAPI tools effectively.
* Managing large DNA sequence datasets efficiently.
* Ensuring correctness and accuracy of codon prediction algorithms.

**Milestones:**
</br>
* Data collection and preprocessing.
* Initial algorithm implementation.
* Integration with Intel OneAPI toolkit.
* Performance optimization and profiling.
* Visualization and reporting.

![codon](https://github.com/21cs043kanivarshini/Prediction-of-DNA-Codon-/assets/110712935/3eb7acb9-0cd3-4e90-85e0-a3f91a2402b5)

This project description outlines the goals, tools, and expected outcomes of a DNA Codon prediction project using Intel OneAPI. It combines bioinformatics with high-performance computing to improve DNA analysis and genomics research

<p align="center">
  <img src="https://user-images.githubusercontent.com/110712935/269548414-097b49ca-039a-4296-89e4-31c9e5d75e85.jpg" width="500" height="300" >
</p>


